### ISSUE TITLE
[0.18.1] TypeError: Symbol.hasInstance is read-only

### ISSUE THREAD

**Issue by wKovacs64** [vitest-dev/vitest#1671]
### Describe the bug

My tests that use `vi.mock` (in this case, mocking out the `ora` module) started failing as of vitest v0.18.1 with the following error:

```
TypeError: Cannot assign to read only property 'Symbol(Symbol.hasInstance)' of function 'function(...o) {
    if (e.called = !0, e.callCount++, e.calls.push(o), e.next) {
      let [s, l] = e.next;
  ...<omitted>... }'
```

In the reproduction (below), it presents itself as simply: `TypeError: Symbol.hasInstance is read-only`

I'm assuming this is related to PR #1648 but I'm not sure if I need to update my usage to comply with the intent of the PR or what.

### Reproduction

https://stackblitz.com/edit/vitest-dev-vitest-ht4a9x?file=test/mock.test.ts&view=editor

### System Info

```shell
System:
    OS: Windows 10 10.0.19044
    CPU: (16) x64 Intel(R) Core(TM) i9-9900K CPU @ 3.60GHz
    Memory: 17.74 GB / 31.92 GB
  Binaries:
    Node: 16.16.0 - C:\Program Files\nodejs\node.EXE
    Yarn: 1.22.18 - ~\AppData\Roaming\npm\yarn.CMD
    npm: 8.12.2 - C:\Program Files\nodejs\npm.CMD
  Browsers:
    Chrome: 103.0.5060.114
    Edge: Spartan (44.19041.1266.0), Chromium (103.0.1264.62)
    Internet Explorer: 11.0.19041.1566
  npmPackages:
    vitest: 0.18.1 => 0.18.1
```

### Used Package Manager

npm

### Validations

- [X] Follow our [Code of Conduct](https://github.com/vitest-dev/vitest/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guidelines](https://github.com/vitest-dev/vitest/blob/main/CONTRIBUTING.md).
- [X] Read the [docs](https://vitest.dev/guide/).
- [X] Check that there isn't [already an issue](https://github.com/vitest-dev/vitest/issues) that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A open a [GitHub Discussion](https://github.com/vitest-dev/vitest/discussions) or join our [Discord Chat Server](https://chat.vitest.dev).
- [X] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

**Comment by sheremet-va** [comment]
@simon-abbott do you have any ideas on how to fix this?

**Comment by spacedawwwg** [comment]
@sheremet-va @simon-abbott 

could it be that `type Key = string | symbol` needs exporting? 
https://github.com/vitest-dev/vitest/blob/fe69f433eefac098c4f4da8d3ccd9497c6059dbf/packages/vitest/src/runtime/mocker.ts#L31

**Comment by spacedawwwg** [comment]
Still an issue in 0.19.0
<img width="1069" alt="image" src="https://user-images.githubusercontent.com/1691867/180641996-b58c1e17-ece8-408f-8750-6610e3394ace.png">

**Comment by dojineko** [comment]
Reproduced it with 0.19.1.
I used a workaround like this. :)

```ts
vi.mock('something', async () => ({
  // @ts-ignore for mocking
  ...(await vi.importActual('something')),
}))
```

**Comment by sheremet-va** [comment]
Please, don't post comments like "still not working". When someone will work on a fix, it will be displayed here.

**Comment by nieyuyao** [comment]
When the `newContainer[property]` is a function, is there no need to continue recursion?

https://github.com/vitest-dev/vitest/blob/57c2367196b3fd978a04fa38ebdac7a5b6ef9b16/packages/vitest/src/runtime/mocker.ts#L213-L217

**Comment by simon-abbott** [comment]
> @simon-abbott do you have any ideas on how to fix this?

Hmm. I know roughly what is causing this, but I'll need a little bit of time to be able to fix it. I'll try to take a look this evening.

**Comment by simon-abbott** [comment]
> When the `newContainer[property]` is a function, is there no need to continue recursion?

Not recursing into functions was the _cause_ of #1523, since Classes in javascript are just functions with special properties.

**Comment by simon-abbott** [comment]
Figured out a fix! #1786.
